client/core/bookie.go

```diff
@@ -743,17 +788,58 @@ func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message)
 	}
 	dc.tradeMtx.RUnlock()
 
-	// Clear the book.
-	book.send(&BookUpdate{
-		Action:   FreshBookAction,
-		Host:     dc.acct.host,
-		MarketID: sp.MarketID,
-		Payload: &MarketOrderBook{
+	func() { // for clean mutex scoping
+		// Locking on dc.booksMtx is currently the only way to ensure the "book"
+		// notification gets sent first, before we start applying any update
+		// notifications coming from server (that can potentially happen once
+		// booky.Reset func returns). For this particular "trade suspension" case
+		// we shouldn't receive any update notifications with seq number greater
+		// than seq in new order book snapshot, and those older will be dropped
+		// anyway (because their seq number is out of date).
+		// But, concurrent Reset calls can result in discrepancy between order
+		// book contents and seq number - so, have to lock dc.booksMtx to avoid
+		// at least that.
+		dc.booksMtx.Lock()
+		defer dc.booksMtx.Unlock()
+
+		book := dc.books[sp.MarketID]
+		if book == nil {
+			err = fmt.Errorf("no order book found with market id '%s'", sp.MarketID)
+			return
+		}
+
+		err = book.Reset(&msgjson.OrderBook{
+			MarketID: sp.MarketID,
+			Seq:      sp.Seq,        // forces seq reset, but should be in seq with previous
+			Epoch:    sp.FinalEpoch, // unused?
+			// Orders is nil
+			// BaseFeeRate = QuoteFeeRate = 0 effectively disables the book's fee
+			// cache until an update is received, since bestBookFeeSuggestion
+			// ignores zeros.
+		})
+		// Return any non-nil error, but still revoke purged orders (done already at
+		// this point) and clear the book.
```

Comment:
Is it OK to **revoke all active orders** before we **Reset** order book ? We'd want to do it to not hold `dc.booksMtx` longer than necessary.